v4/core/utils.go

```diff
@@ -179,3 +180,65 @@ func PrettyPrint(result interface{}, resultName string) {
 func GetCurrentTime() int64 {
 	return time.Now().Unix()
 }
+
+// ConvertSlice Marshals 'slice' to a json string, performs
+// string manipulation on the resulting string, and converts
+// the string to a '[]string'. If 'slice' is nil, not a 'slice' type,
+// or an error occurred during conversion, an error will be returned
+func ConvertSlice(slice interface{}) (s []string, err error) {
+	inputIsSlice := false
+
+	if IsNil(slice) {
+		err = fmt.Errorf(ERRORMSG_NIL_SLICE)
+		return
+	}
+
+	// Reflect on 'slice' to validate the input is in fact a slice
+	rResultType := reflect.TypeOf(slice)
+
+	switch rResultType.Kind() {
+	case reflect.Slice:
+		inputIsSlice = true
+	default:
+	}
+
+	// If it's not a slice or if it's an empty slice, just return
+	if !inputIsSlice || reflect.ValueOf(slice).Len() == 0 {
+		err = fmt.Errorf("the 'slice' parameter must be a non-empty slice")
```

Comment:
Also, my intuition tells me that we might have to treat an empty slice as valid input, which would result in us simply returning an empty string slice 's'.  This will be determined by whether or not the generated code will in fact end up calling this function for an empty slice.  Think of an optional query parameter that is a Date slice, for example.  We'll need to determine the precise code that will be generated for a scenario like that in order to know whether we could potentially pass in an empty input slice.
My suggestion is to hold off on merging this PR until after the generator changes are at least mostly finished when you'll know exactly what will be generated in those cases.

Reply:
After looking at the generated code for go, I think for optional query params, the nil check we do before adding it to the operation's query should check to make sure that param was at least initialized before we call the new function. Either way, I've updated the function to return an empty string slice in the case the parameter is an empty slice